libraries/shared/watcher/event_watcher_test.go

```diff
@@ -208,5 +208,15 @@ var _ = Describe("Event Watcher", func() {
 
 			Expect(err).To(MatchError(fakes.FakeError))
 		})
+
+		It("doesn't panic if one of the go routines errors and closes the err channel", func() {
+			extractor.ExtractLogsErrors = []error{nil, errExecuteClosed}
```

Comment:
I wonder if this mock should also have a second error it attempts to send so that we're likelier to get a `panic: send on closed channel` if extraction continues after delegate has trigger a send on the quit chan